rust/src/builders/tx_builder.rs

```diff
@@ -944,28 +944,60 @@ impl TransactionBuilder {
             // we make that dummy exunits set a dummy script_data_hash to ensure the tx fails if submitted to a node
             false => match has_dummy_exunit {
                 true =>  Some(ScriptDataHash::from([0u8; ScriptDataHash::BYTE_COUNT])),
-                false =>  calc_script_data_hash(
-                    &redeemers,
-                    &self.witness_builders.witness_set_builder.get_plutus_datum(),
-                    &self.config.costmdls,
-                    &Languages(
+                false => {  
+                    let mut languages = 
                         self.witness_builders.witness_set_builder.scripts
                             .values()
                             .fold(
                                 BTreeSet::<Language>::new(),
-                                |mut langs, script| {
-                                    if let Some(lang) = &Option::<Language>::from(script) {
-                                        langs.insert(*lang);
+                                    |mut langs, script| {
+                                        if let Some(lang) = &Option::<Language>::from(script) {
+                                            langs.insert(*lang);
+                                            langs
+                                        } else {
+                                            langs
+                                        }
+                                    }
+                                );
+                        if let Some(reference_inputs) = self.reference_inputs.clone() {
```

Comment:
I think the indentation on everything after here is off by one

Reply:
@SebastienGllmt done

Reply:
@SebastienGllmt done